jakarta.jdt/org.eclipse.lsp4jakarta.jdt.core/src/main/java/org/eclipse/lsp4jakarta/jdt/core/jsonp/JsonpDiagnosticCollector.java

```diff
@@ -32,7 +42,40 @@ public void completeDiagnostic(Diagnostic diagnostic) {
 
     @Override
     public void collectDiagnostics(ICompilationUnit unit, List<Diagnostic> diagnostics) {
+        if (unit == null) {
+            return;
+        }
+        List<MethodInvocation> allMethodInvocations = ASTUtils.getMethodInvocations(unit);
+        List<MethodInvocation> createPointerInvocations = allMethodInvocations.stream()
+                .filter(this::isCreatePointerInvocation).collect(Collectors.toList());
+        for (MethodInvocation m: createPointerInvocations) {
+            Expression arg = (Expression) m.arguments().get(0);
+            if (isInvalidArgument(arg)) {
+                String msg = "createPointer target must be a sequence of '/' prefixed tokens or an emtpy String";
+                try {
+                    Range range = JDTUtils.toRange(unit, arg.getStartPosition(), arg.getLength());
+                    Diagnostic diagnostic = new Diagnostic(range, msg);
+                    diagnostic.setCode(JsonpConstants.DIAGNOSTIC_CODE_CREATE_POINTER);
+                    completeDiagnostic(diagnostic);
+                    diagnostics.add(diagnostic);
+                } catch (JavaModelException e) {
+                    JakartaCorePlugin.logException("Cannot calculate diagnostics", e);
+                }
+            }
+        }
+    }
 
+    public boolean isCreatePointerInvocation(MethodInvocation m) {
+        return m.getName().getIdentifier().equals(JsonpConstants.CREATE_POINTER) && m.arguments().size() == 1;
     }
 
+    public boolean isInvalidArgument(Expression arg) {
+        if (arg instanceof StringLiteral) {
+            String argValue = ((StringLiteral)arg).getLiteralValue();
+            if (!(argValue.isEmpty() || argValue.matches("^(\\/[^\\/]+)+$"))) {
```

Comment:
Just a question: should an exception be thrown if there is no argument to createPointer()?

Reply:
I don't think so because if there is no argument, the method `isCreatePointerInvocation` returns false. And the diagnostic would not be created in the first place in `collectDiagnostics` as it is not recognized as a valid createPointer invocation.